src/instruction.rs

```diff
@@ -10,53 +10,53 @@ use solana_program::{
 /// Instructions supported by the Paladin Rewards program.
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum PaladinRewardsInstruction {
-    /// Configures staker rewards for a mint that has been configured with the
-    /// rewards program as a transfer hook program.
+    /// Configures a holder rewards pool for a mint that has been configured
+    /// with the rewards program as a transfer hook program.
     ///
     /// This instruction will:
     ///
-    /// - Initialize a staker rewards account.
+    /// - Initialize a holder rewards pool account.
     /// - Initialize the required accounts for the transfer hook.
     ///
     /// Accounts expected by this instruction:
     ///
-    /// 0. `[w]` Staker rewards account.
+    /// 0. `[w]` Holder rewards pool account.
     /// 1. `[w]` Transfer hook extra account metas account.
     /// 2. `[ ]` Token mint.
-    /// 3. `[ ]` Token mint.
-    /// 4. `[s]` Mint authority.
-    /// 5. `[ ]` System program.
-    InitializeStakerRewards,
-    /// Moves SOL rewards to the following parties:
-    ///
-    /// - 1%  Piggy bank.
-    /// - 4%  Staker rewards.
-    /// - 5%  Holder rewards.
-    /// - 90% Leader who produces the block.
+    /// 3. `[s]` Mint authority.
+    /// 4. `[ ]` System program.
+    InitializeHolderRewardsPool,
+    /// Moves SOL rewards to the holder rewards pool and updates the total.
     ///
     /// Accounts expected by this instruction:
     ///
-    /// 0. `[w]` Staker rewards account.
-    /// 1. `[w]` Holder rewards account.
-    /// 2. `[w]` Piggy bank account.
-    /// 3. `[w]` Leader account.
-    DistributeRewards,
+    /// 0. `[w]` Holder rewards pool account.
+    /// 1. `[w, s]` Payer account.
+    /// 2. `[ ]` System program.
+    DistributeRewards(u64),
```

Comment:
I still prefer reversing 0 and 1 so this works like `transfer`, ie source then recipient